### PR TITLE
CI: initial configuration

### PR DESCRIPTION
**Pull request by tarcieri**
Configures GitHub Actions to run clippy, check rustdoc, run rustfmt, and run the test suite

### CODE REVIEW

**Comment by tarcieri** [comment]
@stevefan1999-personal we generally use a stock rustfmt configuration, particularly so it doesn't involve nightly

**Comment by tarcieri** [comment]
Hmm, seeing some test failures I wasn't seeing locally

File: tests/badssl.rs [hunk 1]
```diff
@@ -40,7 +40,7 @@ async fn test_badssl_tls12(uri: &str, expected: Result<(), ()>) {
 #[test_case("https://untrusted-root.badssl.com/", Err(()); "test untrusted root")]
 #[test_case("https://wrong.host.badssl.com/", Err(()); "test wrong host")]
 #[test_case("https://no-sct.badssl.com/", Err(()); "test Signed Certificate Timestamp")] // NET::ERR_CERTIFICATE_TRANSPARENCY_REQUIRED
-#[test_case("https://pinning-test.badssl.com/", Err(()); "test pinning test")] // NET::ERR_SSL_PINNED_KEY_NOT_IN_CERT_CHAIN
+#[cfg_attr(TODO, test_case("https://pinning-test.badssl.com/", Err(()); "test pinning test"))] // NET::ERR_SSL_PINNED_KEY_NOT_IN_CERT_CHAIN
```
[the remark this answers is not in the document]
Comment (stevefan1999-personal) [reply tests/badssl.rs:43]: Yes. I think pinning is broken again in rustls

**Comment by stevefan1999-personal** [comment]
> @stevefan1999-personal we generally use a stock rustfmt configuration, particularly so it doesn't involve nightly

Sure, just delete the configs that needed nightly for now

File: .github/workflows/rustls-rustcrypto.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+#        toolchain:
+#          - stable
+#    runs-on: ubuntu-latest
+#    steps:
+#      - uses: actions/checkout@v4
+#      - uses: RustCrypto/actions/cargo-cache@master
+#      - uses: dtolnay/rust-toolchain@master
+#        with:
+#          toolchain: ${{ matrix.toolchain }}
+#      - run: cargo test
```
Comment (tarcieri) [inline .github/workflows/rustls-rustcrypto.yml:77]: Well, this is less-than-ideal, but I think having the other jobs is a good start.

I'm going to merge this and open a followup which uncomments this workflow so we can see if we can get it (or at least a subset thereof) working in CI.

**Pull request merged**